### DISCUSSION TITLE
Arduino IDE sdkconfig.h

### DISCUSSION THREAD

**Discussion by Mistborn21** [earlephilhower/arduino-pico#1785] [Q&A]
Hello!
I'm working on making a short bit of proof of concept code, but I can't do anything with a pico w on Arduino IDE as it keeps asking for me to #include "sdkconfig.h". I've been looking around on the web and on this repository and didn't see any sdkconfig. I tried to manually add the  pico_sdk_import, but that still gave me the same result. Attached is my code, but I had the same error when doing a base example from Arduino IDE itself with my board set to pico w. 
Thanks!

Error message:

FQBN: rp2040:rp2040:rpipicow
Using board 'rpipicow' from platform in folder: C:\Users\asher\AppData\Local\Arduino15\packages\rp2040\hardware\rp2040\3.6.0
Using core 'rp2040' from platform in folder: C:\Users\asher\AppData\Local\Arduino15\packages\rp2040\hardware\rp2040\3.6.0

Detecting libraries used...
C:\Users\asher\AppData\Local\Arduino15\packages\rp2040\tools\pqt-gcc\2.1.0-a-d3d2e6b/bin/arm-none-eabi-g++ -I C:\Users\asher\AppData\Local\Temp\arduino\sketches\87FE3C3A8252FF2A4EF70D0E85C11F69/core -c -Werror=return-type -Wno-psabi -DCFG_TUSB_MCU=OPT_MCU_RP2040 -DUSBD_PID=0xf00a -DUSBD_VID=0x2e8a -DUSBD_MAX_POWER_MA=250 -DUSB_MANUFACTURER="Raspberry Pi" -DUSB_PRODUCT="Pico W" -DPICO_CYW43_ARCH_THREADSAFE_BACKGROUND=1 -DCYW43_LWIP=1 -DLWIP_IPV6=0 -DLWIP_IPV4=1 -DLWIP_IGMP=1 -DLWIP_CHECKSUM_CTRL_PER_NETIF=1 -DARDUINO_VARIANT="rpipicow" -DTARGET_RP2040 -DPICO_FLASH_SIZE_BYTES=2097152 -march=armv6-m -mcpu=cortex-m0plus -mthumb -ffunction-sections -fdata-sections -fno-exceptions -DARM_MATH_CM0_FAMILY -DARM_MATH_CM0_PLUS -iprefixC:\Users\asher\AppData\Local\Arduino15\packages\rp2040\hardware\rp2040\3.6.0/ @C:\Users\asher\AppData\Local\Arduino15\packages\rp2040\hardware\rp2040\3.6.0/lib/platform_inc.txt -IC:\Users\asher\AppData\Local\Arduino15\packages\rp2040\hardware\rp2040\3.6.0/include -fno-rtti -std=gnu++17 -g -pipe -w -x c++ -E -CC -DF_CPU=133000000L -DARDUINO=10607 -DARDUINO_RASPBERRY_PI_PICO_W -DBOARD_NAME="RASPBERRY_PI_PICO_W" -DARDUINO_ARCH_RP2040 -Os -DWIFICC=CYW43_COUNTRY_WORLDWIDE -IC:\Users\asher\AppData\Local\Arduino15\packages\rp2040\hardware\rp2040\3.6.0\cores\rp2040 -IC:\Users\asher\AppData\Local\Arduino15\packages\rp2040\hardware\rp2040\3.6.0\variants\rpipicow C:\Users\asher\AppData\Local\Temp\arduino\sketches\87FE3C3A8252FF2A4EF70D0E85C11F69\sketch\sketch_oct22a.ino.cpp -o nul
Alternatives for PS4Controller.h: [PS4Controller@1.0.6]
ResolveLibrary(PS4Controller.h)
  -> candidates: [PS4Controller@1.0.6]
C:\Users\asher\AppData\Local\Arduino15\packages\rp2040\tools\pqt-gcc\2.1.0-a-d3d2e6b/bin/arm-none-eabi-g++ -I C:\Users\asher\AppData\Local\Temp\arduino\sketches\87FE3C3A8252FF2A4EF70D0E85C11F69/core -c -Werror=return-type -Wno-psabi -DCFG_TUSB_MCU=OPT_MCU_RP2040 -DUSBD_PID=0xf00a -DUSBD_VID=0x2e8a -DUSBD_MAX_POWER_MA=250 -DUSB_MANUFACTURER="Raspberry Pi" -DUSB_PRODUCT="Pico W" -DPICO_CYW43_ARCH_THREADSAFE_BACKGROUND=1 -DCYW43_LWIP=1 -DLWIP_IPV6=0 -DLWIP_IPV4=1 -DLWIP_IGMP=1 -DLWIP_CHECKSUM_CTRL_PER_NETIF=1 -DARDUINO_VARIANT="rpipicow" -DTARGET_RP2040 -DPICO_FLASH_SIZE_BYTES=2097152 -march=armv6-m -mcpu=cortex-m0plus -mthumb -ffunction-sections -fdata-sections -fno-exceptions -DARM_MATH_CM0_FAMILY -DARM_MATH_CM0_PLUS -iprefixC:\Users\asher\AppData\Local\Arduino15\packages\rp2040\hardware\rp2040\3.6.0/ @C:\Users\asher\AppData\Local\Arduino15\packages\rp2040\hardware\rp2040\3.6.0/lib/platform_inc.txt -IC:\Users\asher\AppData\Local\Arduino15\packages\rp2040\hardware\rp2040\3.6.0/include -fno-rtti -std=gnu++17 -g -pipe -w -x c++ -E -CC -DF_CPU=133000000L -DARDUINO=10607 -DARDUINO_RASPBERRY_PI_PICO_W -DBOARD_NAME="RASPBERRY_PI_PICO_W" -DARDUINO_ARCH_RP2040 -Os -DWIFICC=CYW43_COUNTRY_WORLDWIDE -IC:\Users\asher\AppData\Local\Arduino15\packages\rp2040\hardware\rp2040\3.6.0\cores\rp2040 -IC:\Users\asher\AppData\Local\Arduino15\packages\rp2040\hardware\rp2040\3.6.0\variants\rpipicow -Ic:\Users\asher\Documents\Arduino\libraries\PS4Controller\src C:\Users\asher\AppData\Local\Temp\arduino\sketches\87FE3C3A8252FF2A4EF70D0E85C11F69\sketch\sketch_oct22a.ino.cpp -o nul
In file included from C:\Users\asher\AppData\Local\Temp\.arduinoIDE-unsaved2023922-17524-1bpag4v.uosc\sketch_oct22a\sketch_oct22a.ino:1:
E:\SchoolStuff\CollegeStuff\Year4\SeniorProject\Arduino Code\pico_sdk_import.cmake:1:3: error: invalid preprocessing directive #This
    1 | # This is a copy of <PICO_SDK_PATH>/external/pico_sdk_import.cmake
      |   ^~~~
E:\SchoolStuff\CollegeStuff\Year4\SeniorProject\Arduino Code\pico_sdk_import.cmake:3:3: error: invalid preprocessing directive #This
    3 | # This can be dropped into an external project to help locate this SDK
      |   ^~~~
E:\SchoolStuff\CollegeStuff\Year4\SeniorProject\Arduino Code\pico_sdk_import.cmake:4:3: error: invalid preprocessing directive #It
    4 | # It should be include()ed prior to project()
      |   ^~
In file included from C:\Users\asher\AppData\Local\Temp\.arduinoIDE-unsaved2023922-17524-1bpag4v.uosc\sketch_oct22a\sketch_oct22a.ino:4:
c:\Users\asher\Documents\Arduino\libraries\PS4Controller\src/ps4_int.h:4:10: fatal error: sdkconfig.h: No such file or directory
    4 | #include "sdkconfig.h"
      |          ^~~~~~~~~~~~~
compilation terminated.
Alternatives for sdkconfig.h: []
ResolveLibrary(sdkconfig.h)
  -> candidates: []

Using library PS4Controller at version 1.0.6 in folder: C:\Users\asher\Documents\Arduino\libraries\PS4Controller 
exit status 1

Compilation error: exit status 1

Code:

#include "E:\SchoolStuff\CollegeStuff\Year4\SeniorProject\Arduino Code\pico_sdk_import.cmake"
#include <PS4Controller.h>
#include <ps4.h>
#include <ps4_int.h>

void setup() {
  PS4.begin();
}

void loop() {

  if (PS4.isConnected()){
    if (PS4.Square()){
        digitalWrite(LED_BUILTIN, HIGH);  // turn the LED on (HIGH is the voltage level)
        delay(1000);                      // wait for a second
        digitalWrite(LED_BUILTIN, LOW);   // turn the LED off by making the voltage LOW
        delay(1000);
    }
  }

}

**Comment by maxgerhardt** [comment]
1. Please format your post properly using tripple-backticks for [code blocks](https://www.markdownguide.org/extended-syntax/#fenced-code-blocks)
2.  The "PS4Library" you're including is literally only for ESP32 ([see code](https://github.com/aed3/PS4-esp32/blob/master/src/ps4_int.h#L4-L7)). The code is not capable of being compiled for a Raspberry Pico W, even though it also has Bluetooth. Adapting the library for that would require code changes.
3. There are projects out there using a Pico W with a PS4 controller, using the bluetooth stack API and the verbatim Pico SDK: https://github.com/usedbytes/picow_ds4. You might be able to adapt the code for Arduino.
4. For other RP2040 boards that use a NINA module for the WiFi + Bluetooth communication instead, there is the Arduino library https://github.com/ricardoquesada/bluepad32-arduino. This does not work on Pico W (which has a Cypress WiFi/module). If you want the least amount of trouble / work, I'd recommend just buying a Arduino Nano RP2040 Connect and using that tested library.
